calyx-ir/src/control.rs

```diff
@@ -175,8 +263,20 @@ pub enum Control {
     Enable(Enable),
     /// Control statement that does nothing.
     Empty(Empty),
-    /// Runs the control for a static group.
-    StaticEnable(StaticEnable),
+    /// Static Control
+    Static(StaticControl),
+}
+
+/// Control AST nodes.
+#[derive(Debug)]
+pub enum StaticControl {
```

Comment:
One problem with skipping `invoke` is that "control promotion" pass cannot promote a sequence of invokes which are likely to occur since we have `group2invoke`

Reply:
One thing we were thinking is that we could (in infer-static-timing/ "group promotion") turn dynamic invokes into static enables/groups, which would help w/ "control promotion". 
But there is still the consequence that `group2invoke` will only transform dynamic groups-> dynamic invokes. We can't transform static groups -> static invokes. @paili0628 not sure if you also had something else in mind?

Reply:
Right! There is a somewhat important "pass ordering" question here: is `group2invoke` allowed to run in presence of `static group`s or are they supposed to be eliminated by that point. 

The turning of dynamic invokes into static enables feels somewhat like the job of `compile-invoke`. That is, it's not clear to me what the benefit of doing that is apart from keeping the IR smaller but I think in general you want frontend programs to be able to generate `static invoke` and have the compiler use that information for optimizations

Reply:
Ok, we decided to add a `static invoke`